src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperBlockRewriter.cs

```diff
@@ -61,10 +63,14 @@ internal static class TagHelperBlockRewriter
                 {
                     SourceLocation? errorLocation = null;
 
-                    // Check if it's a bound attribute that is minimized or if it's a bound non-string attribute that
-                    // is null or whitespace.
-                    if ((result.IsBoundAttribute && result.AttributeValueNode == null) ||
+                    // Check if it's a non-boolean bound attribute that is minimized or if it's a bound
+                    // non-string attribute that is null or whitespace.
+                    var isMinimizedBoundAttribute = result.IsBoundAttribute && result.AttributeValueNode == null;
+                    var isValidMinimizedAttribute = allowMinimizedBooleanTagHelperAttributes && result.IsBoundBooleanAttribute;
+                    if ((isMinimizedBoundAttribute &&
+                        !isValidMinimizedAttribute) ||
                         (result.IsBoundNonStringAttribute &&
+                        result.AttributeValueNode != null &&
```

Comment:
This is kind of ugly. Can we split the below error message into two? 
`Attribute '{0}' on tag helper element '{1}' requires a value. Tag helper bound attributes of type '{2}' cannot be empty or contain only whitespace.`
We can have one for null and one for whitespace. That will make the code here simpler. Thoughts?

Reply:
Eh, can just leave the error message as is. You should factor your if statement into two if's for more clarity:
1. if feature is disabled - act like the old way
2. else act like new way

Reply:
Actually, can't you move this error entirely to the intermediate phase? It's crazy that we did these types of checks here to begin with.

Reply:
Yup. I like that. If I can move this error to lowering phase, this will get simplified.